.github/workflows/check_code_format.yml

```diff
@@ -14,8 +14,29 @@ jobs:
         cd dev-tools/ && ./bootstrap
     - name: Check all .cc , .h, .proto files for correct code format
       run: chmod +x dev-tools/check_style.sh && dev-tools/./check_style.sh
+
     - name: Install buildifier for .bzl files
       run: brew install buildifier
+
     - name: Check all .bzl, .bazel files for correct code format
-      run: chmod +x dev-tools/check_style_bzl.sh && dev-tools/./check_style_bzl.sh    
+      run: chmod +x dev-tools/check_style_bzl.sh && dev-tools/./check_style_bzl.sh
+
+    - name: Generate compile_commands.json
+      run: chmod +x dev-tools/generate_compile_commands.sh && dev-tools/./generate_compile_commands.sh       
+
+    - name: Use clang-tidy-review action to review any PR on code style
+      uses: ZedThree/clang-tidy-review@v0.7.0:
+      id: review
+      with:
+        # List of clang-tidy checks.
+        clang_tidy_checks: '-*,performance-*,readability-*,bugprone-*,clang-analyzer-*,cppcoreguidelines-*,mpi-*,misc-*,-readability-implicit-bool-conversion,-readability-qualified-auto,-cppcoreguidelines-owning-memory,-cppcoreguidelines-avoid-magic-numbers,-readability-magic-numbers,-readability-static-accessed-through-instance,-cppcoreguidelines-pro-type-static-cast-downcast,-bugprone-suspicious-enum-usage,-readability-uppercase-literal-suffix,-cppcoreguidelines-avoid-non-const-global-variables,-cppcoreguidelines-pro-bounds-array-to-pointer-decay,-misc-no-recursion,-readability-isolate-declaration,-cppcoreguidelines-pro-type-vararg,-misc-non-private-member-variables-in-classes,-performance-no-automatic-move,-readability-avoid-const-params-in-decls,-cppcoreguidelines-non-private-member-variables-in-classes'
+    
+    - name: Passed C++ style guide checks
+      # Doesn't work for now the if statement below.  
+      if: steps.static_analysis.outputs.total_comments > 0
+      run: exit 1
```

Comment:
What do we need to do to make this work? We don't want to include code that doesn't work!

Reply:
I believe the comment to be miss leading here. This step probably works as intended, but the previous step, `static_analysis`, doesn't.
As the previous step always has status code 0, even if there are `clang-tidy` errors, this step turns it in to a fail if there is output written.
The previous step however is the part I think is not working. As I am understanding it it currently doesn't pick up the source and header files, checks 0 files, makes no output and this check here does not (and likely should not) catch that.

I'm sure @ArthurBandaryk  can clarify in case I missed something here.